src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -6,11 +6,20 @@
 #include <string.h>
 #include <vector>
 
-#include "config.h"
+#include "icushim.h"
 #include "locale.hpp"
 #include "holders.h"
 #include "errors.h"
 
+#ifndef FEATURE_FIXED_ICU_VERSION
+// The System.Globalization.Native.so is built in mode when it can use any supported version
+// of ICU installed on the target machine and not just the one it was built with. So we can 
+// only use stable APIs and the UDAT_STANDALONE_SHORTER_WEEKDAYS doesn't belong to those.
+// We also want to support ICU versions where the UDAT_STANDALONE_SHORTER_WEEKDAYS was not 
+// defined yet.
+#undef HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS
```

Comment:
I guess my question is why `HAVE_SET_MAX_VARIABLE` and `HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS` have to be treated differently if we make the changes you describe.  `UDAT_STANDALONE_SHORTER_WEEKDAYS` was not present in ICU 50 (it went stable in ICU 51) and so the configuration test will fail.  If we try to take version of System.Globalization.Native.so built on ICU 51 and run it on a platform with ICU 50 or earlier, our loop to find a library to dlopen would fail, because we stop searching at `U_ICU_VERSION_MAJOR_NUM` right?